core/src/main/java/org/infinispan/manager/impl/LocalClusterExecutor.java

```diff
@@ -52,20 +52,20 @@ Address getMyAddress() {
    @Override
    public void execute(Runnable command) {
       // We ignore time out since user can't ever even respond to it, so no reason to create extra fluff
-      localExecutor.execute(command);
+      blockingManager.runBlocking(command, getClass().getSimpleName() + "-execute");
    }
 
    @Override
    public CompletableFuture<Void> submit(Runnable command) {
       CompletableFuture<Void> future = new CompletableFuture<>();
-      localExecutor.execute(() -> {
-         try {
-            command.run();
-            future.complete(null);
-         } catch (Throwable t) {
-            future.completeExceptionally(t);
-         }
-      });
+      blockingManager.runBlocking(command, getClass().getSimpleName() + "-submit")
+            .whenComplete((ignore, t) -> {
+               if (t != null) {
+                  future.completeExceptionally(t);
+               } else {
+                  future.complete(null);
+               }
+            });
```

Comment:
can be simplified: `CompletableFuture<Void> future = blockingManager.runBlocking(...).toCompletableFuture()`

Reply:
Argh, I had to revert this to use the `.whenComplete`. Otherwise, consumers receive a `CompletionException` instead of the actual exception.